utils/bundleHelper.js

```diff
@@ -3,13 +3,25 @@ import * as fs from "fs";
 
 let filePath = process.argv[2];
 let outputFileName = process.argv[3];
+
 $RefParser.bundle(filePath).then(async (schema) => {
+
     let jsonSchema = decodeURI(JSON.stringify(schema, null, 2));
     jsonSchema = jsonSchema.replaceAll('%24', '$');
-    fs.writeFile(outputFileName, jsonSchema, 'utf8', (err) => {
-        if (err) throw err;
-        console.log('The file has been saved!');
-    });
+
+    if(outputFileName === "application.schema.json") {
+        fs.writeFile(outputFileName, JSON.stringify(jsonSchema, null, 2), 'utf8', (err) => {
+              if (err) throw err;
+              console.log('The file has been saved!');
+        });
+    } else {
+        let jsonSchemaObject = JSON.parse(jsonSchema);
+        jsonSchemaObject['properties']['parameters']['items']['properties']['parameter_state_type']['$ref'] = '#/properties/inputs/items/properties/signal_types/items';
```

Comment:
nitpick: I would just suggest to add a comment here to explain why this is needed, for future reference